### PR TITLE
[Storage] Drop support for Python 2.7 and 3.6

### PR DESCRIPTION
**Pull request by jalauzon-msft**
Dropping support for Python 2.7 and 3.6 with the following steps for each package:

- Modifying `setup.py`.
- Modifying `dev_requirements.txt`.
- Modifying `mypy.ini` to use Python 3.7.
- Modifying package `README.md`.
- Modifying `samples/README.md`.
- Removing `setup.cfg`.

### CODE REVIEW

File: sdk/storage/azure-storage-queue/setup.py [hunk 1]
```diff
@@ -62,11 +47,7 @@
     classifiers=[
         "Development Status :: 4 - Beta",
         'Programming Language :: Python',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/setup.py:50]: "Programming Language :: Python :: 3 :: Only"

File: sdk/storage/azure-storage-blob-changefeed/README.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+## _Disclaimer_
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob-changefeed/README.md:1]: Could we add a note to the changelog for the next release for each of these packages?

Comment (jalauzon-msft) [reply sdk/storage/azure-storage-blob-changefeed/README.md:1]: Good idea. Done.

File: sdk/storage/azure-storage-queue/setup.py (hunk 1 above)
Comment (jalauzon-msft) [reply sdk/storage/azure-storage-queue/setup.py:51]: Oops, missed one. Good catch.

**Comment by tasherif-msft** [comment]
Looks good, we should also remove test checks and import checks for python 2 for example:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-blob/tests/test_container.py#L939
and 
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py#L15-L18

**Review by tasherif-msft — APPROVED** [review]

**Comment by jalauzon-msft** [comment]
Opened a follow-up issue to remove Python 2 specific checks from the code. #22427

**Pull request merged**